module/applications/actor/base-sheet.mjs

```diff
@@ -1076,6 +1091,35 @@ export default class ActorSheet5e extends ActorSheet {
         return false;
       }
     }
+
+    // Adjust the preparation mode of a leveled spell depending on the section on which it is dropped.
+    if ( itemData.type === "spell" && itemData.system.level != 0 ) {
```

Comment:
It might be worth extracting this into a dedicated method (named something like `onDropSpell`) and then use the same calling pattern as [line 1083](https://github.com/foundryvtt/dnd5e/blob/f3fe4c72285e62a3a9eb8cca2b54541a43d5aa52/module/applications/actor/base-sheet.mjs#L1083..L1084).

Reply:
I agree, it grew a bit big.